### DISCUSSION TITLE
delimited-dsl where closing delimiter is also an escape leader

### DISCUSSION THREAD

**Discussion by dancingbug** [foonathan/lexy#62] [Q&A] [answered]
How to write a delimited-dsl closing delimiter that is the same as an escape leader?

suppose for sake of illustration that in the object language to be parsed, strings were delimited by `%`
Then I would want to match delimited strings as follows:
- `"%one two%" -->  "one two"`
- `"%one %% two%" -->  "one % two"`

In Pascal, for instance, the basic syntax of a string is delimited with single quotes.
If you want to escape a single quote inside such a string, you type it twice. 

What is the recommended way to handle this?

here is a snippet that did not work
```
struct pct_quoted {
  static constexpr auto quote_sym = lexy::symbol_table<char>.map<'%'>('%');
  static constexpr auto rule = [] {
    auto delim = dsl::percent_sign;
    auto escape = dsl::escape(delim).symbol<quote_sym>();
    auto pct_q = dsl::delimited(delim, delim);
    return pct_q(dsl::ascii::character,escape);
  }();
};

void test_pct_quot() {
  auto input = lexy::zstring_input("%one%%two%");

  lexy::trace<pct_quoted>(stdout,input);
};
```

```
 1:  1: pct_quoted:
 1:  1: - literal: %
 1:  2: - token: one
 1:  5: - literal: %
 1:  6: - finish
```

Thank you for this nice library.

**Comment by foonathan** [comment, marked as answer]
A simple fix would be to change the closing delimiter to match `%` but not `%%` using `dsl::not_followed_by`: https://lexy.foonathan.net/playground/?id=xaazc4dMo&mode=tree.
